ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CoveragesPage.tsx

```diff
@@ -72,7 +73,19 @@ function CoveragesPage() {
         Cluster: getFilteredConfig(clusterSearchFilterConfig, ['Name']),
     };
 
-    const fetchProfilesStats = useCallback(() => getComplianceProfilesStats(), []);
+    const fetchProfilesStats = useCallback(async () => {
+        const response = await getComplianceProfilesStats();
+        // Prevents a page shift when only relying on the useEffect below. Otherwise, stats will be finished loading
+        // but selectedProfileStats will be undefined for a moment causing the widget container to not render.
+        if (response) {
+            const profileStats = response.scanStats.find(
+                (profile) => profile.profileName === profileName
+            );
+            setSelectedProfileStats(profileStats);
+        }
```

Comment:
We might need to consider improving `useRestQuery` to handle side effects in cases like this. Since you want to set the state after the response returns, `useRestQuery` could expose a function like `onComplete(response) => void` that allows us to handle such scenarios. This approach would separate concerns, enabling `fetchProfileStats` to focus solely on making the API call through the service function.

I think I created a `useCreateReport` back in the day (before `useRestQuery`) where I did something like this.

```
const { isLoading, error, createReport } = useCreateReport({
        onCompleted: () => {
            clearFormValues();
            history.push(vulnerabilityReportsPath);
        },
});
```

Considering the deadline, I don't think you should attempt to do it now (unless you feel it's feasible). But we can add a comment or track the improvement in Jira for now.

Reply:
I like that idea 👍  I would just want to make sure the `onCompleted` and response changes are applied synchronously, if that makes sense. I originally had this logic in a `useEffect` but that caused a layout shift on the page because the profile stats `data` existed and `isLoading` was false monetarily before `setSelectedProfileStat` state was updated.